packages/core/src/queue/content.js

```diff
@@ -28,7 +24,7 @@ const decode = (rows) => {
   return rows.map(content => ({
     link: parseLink(/** @type {string} */ (content.link)),
     size: /** @type {number} */(Number.parseInt(/** @type {string} */ (content.size))) | 0,
-    source: /** @type {ContentSource[]} */ (content.source),
+    source: /** @type {URL[]} */ (content.source.map((/** @type {string} */ s) => new URL(s))),
```

Comment:
Nit: Malformed URLs will throw an exception during parse, maybe leave it as string and deal with parse in the consumer code ? It also may be a good idea to to validate URL in the encode above.

I fear that one malformed URL may halt the system because all peeks will keep on failing at decode

Reply:
Alternatively decode could also return Result types

Reply:
Yes, so I intend to perform those validations where we receive the URLs. However, I think it might make sense to validate in the queue entry point as well, so adding a validation on `put` that will fail if we receive bad URLs